src/coreclr/tools/Common/Compiler/DependencyAnalysis/ObjectDataBuilder.cs

```diff
@@ -3,17 +3,29 @@
 
 using System;
 using System.Collections.Generic;
+#if !HOST_MODEL
```

Comment:
Would a partial class make sense for this?

Or how much _interesting_ logic are we really sharing? If there's not much interesting logic, we could just have a separate struct named `ObjectDataBuilder` for host model that quacks the same (has the methods we need to compile the shared code). This struct is pretty stable. We sometimes add stuff to it, but the base has been the same for many years.

Reply:
I think it doesn't have much interesting logic so I'll create separate struct and I'll leave comment at the top of ObjectDataBuilder.cs.